doc/subsystems/conf_s11n/conf_s11n.rst

```diff
@@ -0,0 +1,156 @@
+.. _conf_s11n
+
+Runtime configuration system
+############################
+
+Config subsystem is meant for persisting per-device configuration
+and runtime state for modules.
+
+Configuration items are stored as key-value pairs, where both the key and
+the value are expected to be strings. Keys are divided into component
+elements, where packages register their subtree using the first element.
+E.g key ``id/serial`` consists of 2 components, ``id`` and ``serial``.
+Package sys/id registered it's subtree of configuration elements to be
+under ``id``.
+
+There are convenience routines for converting value back and forth
+from string.
+
+Handlers
+********
+
+Config handlers for subtree implement a set of handler functions.
+These are registered using a call to ``conf_register()``.
+
+**ch_get**
+    This gets called when somebody asks for a config element value
```

Comment:
This gets called when asking for a config element value by its name
    using ``conf_get_value()``.